File: backend/src/Shared/Overview/OverviewResponse.cs
namespace Shared
{
    public class OverviewResponse
    {
        public string Type { get; set; }
        public string QueryString { get; set; }

        public TimeModePeriodObject TimeModePeriodObject { get; set; }

        public TimePeriodObject TimePeriodObject { get; set; }

        public OverviewResponse(string type, string queryString, TimeModePeriodObject timeModePeriodObject, TimePeriodObject timePeriodObject)
        {
            Type = type;
            QueryString = queryString;
            TimeModePeriodObject = timeModePeriodObject;
            TimePeriodObject = timePeriodObject;
        }
    }
}

File: frontend/src/types/overviewResponse.ts
interface TimeModePeriodObject {
  name: string;
  timespan: string;
  total: number;
}

interface ThenByObject {
  name: string;
  total: number;
}

interface GroupByObject {
  name: string;
  total: number;
  thenByObjects: ThenByObject[];
}

interface TimePeriodObject {
  groupByObjects: GroupByObject[];
}

export interface OverviewResponse {
  type: string;
  queryString: string;
  timeModePeriodObject: TimeModePeriodObject;
  timePeriodObject: TimePeriodObject;
}
